// @servicetitan/anvil2/dist/token/core/raw.js
export const common = {
	BorderRadiusNone: "0",
	BorderRadiusSmall: "0.1875rem",
	BorderRadiusMedium: "0.375rem",
	BorderRadiusLarge: "0.75rem",
	BorderRadiusXlarge: "1.5rem",
	BorderRadiusCircular: "100%",
	BorderWidthNone: "0rem",
	BorderWidth: "0.0625rem",
	BorderWidthStrong: "0.125rem",
	ChartsMonochrome01Pattern: "solid",
	ChartsMonochrome02Pattern: "solid",
	ChartsMonochrome03Pattern: "striped",
	ChartsMonochrome04Pattern: "solid",
	ChartsCategorical10Pattern: "solid",
	ChartsCategorical01Pattern: "solid",
	ChartsCategorical02Pattern: "solid",
	ChartsCategorical03Pattern: "solid",
	ChartsCategorical04Pattern: "solid",
	ChartsCategorical05Pattern: "solid",
	ChartsCategorical06Pattern: "solid",
	ChartsCategorical07Pattern: "solid",
	ChartsCategorical08Pattern: "solid",
	ChartsCategorical09Pattern: "solid",
	ChartsStatusSuccessPattern: "solid",
	ChartsStatusWarningPattern: "solid",
	ChartsStatusDangerPattern: "solid",
	ChartsStatusNeutralPattern: "striped",
	FocusRingWidth: "0.25rem",
	ShadowSizeFlat: "0rem 0rem 0rem",
	ShadowSizeFloat: "0rem 0.125rem 0.5rem",
	ShadowSizeOverlay: "0rem 0.5rem 1.5rem",
	TypographyParagraphSizeXsmall: "0.75rem",
	TypographyParagraphSizeSmall: "0.875rem",
	TypographyParagraphSize: "1rem",
	TypographyParagraphSizeLarge: "1.25rem",
	TypographyParagraphSizeXlarge: "1.5rem",
	TypographyParagraphFontWeight: 400,
	TypographyParagraphFontFamily: "'Nunito Sans', sans-serif",
	TypographyHeadingSizeXsmall: "0.875rem",
	TypographyHeadingSizeSmall: "1rem",
	TypographyHeadingSize: "1.25rem",
	TypographyHeadingSizeLarge: "1.5rem",
	TypographyHeadingSizeXlarge: "2rem",
	TypographyHeadingFontWeight: 700,
	TypographyHeadingFontFamily: "'Sofia Pro', SofiaPro, sans-serif",
	TypographyLabelSizeXsmall: "0.625rem",
	TypographyLabelSizeSmall: "0.75rem",
	TypographyLabelSize: "0.875rem",
	TypographyLabelSizeLarge: "1rem",
	TypographyLabelSizeXlarge: "1.25rem",
	TypographyLabelFontWeight: 600,
	TypographyLabelFontFamily: "'Nunito Sans', sans-serif"
}
export const light = {
	BackgroundColor: "#ffffff",
	BackgroundColorStrong: "#f7f7f7",
	BackgroundColorStronger: "#eeeeee",
	BackgroundColorStrongest: "#bcbcbd",
	BackgroundColorPrimary: "#0265DC",
	BackgroundColorPrimarySubdued: "#E0F2FF",
	BackgroundColorSuccess: "#007A4D",
	BackgroundColorSuccessSubdued: "#CEF8E0",
	BackgroundColorDanger: "#e13212",
	BackgroundColorDangerSubdued: "#ffece9",
	BackgroundColorWarning: "#ffbe00",
	BackgroundColorDisabled: "#606162",
	BackgroundColorInverted: "#2d2e31",
	BackgroundColorInvertedStrong: "#141414",
	BorderColor: "#949596",
	BorderColorSubdued: "#dfe0e1",
	BorderColorStrong: "#444445",
	BorderColorPrimary: "#0265DC",
	BorderColorSuccess: "#007A4D",
	BorderColorDanger: "#e13212",
	ChartsMonochrome01Color: "#3892F3",
	ChartsMonochrome01Stroke: "transparent",
	ChartsMonochrome02Color: "#08389c",
	ChartsMonochrome02Stroke: "transparent",
	ChartsMonochrome03Color: "#F2F9FF",
	ChartsMonochrome03Stroke: "#3892F3",
	ChartsMonochrome04Color: "#F2F9FF",
	ChartsMonochrome04Stroke: "#3892F3",
	ChartsCategorical10Color: "#D57A5C",
	ChartsCategorical10Stroke: "transparent",
	ChartsCategorical01Color: "#3892F3",
	ChartsCategorical01Stroke: "transparent",
	ChartsCategorical02Color: "#08389c",
	ChartsCategorical02Stroke: "transparent",
	ChartsCategorical03Color: "#0FA2B8",
	ChartsCategorical03Stroke: "transparent",
	ChartsCategorical04Color: "#907AF5",
	ChartsCategorical04Stroke: "transparent",
	ChartsCategorical05Color: "#DE52CE",
	ChartsCategorical05Stroke: "transparent",
	ChartsCategorical06Color: "#C2185B",
	ChartsCategorical06Stroke: "transparent",
	ChartsCategorical07Color: "#759E15",
	ChartsCategorical07Stroke: "transparent",
	ChartsCategorical08Color: "#AC6400",
	ChartsCategorical08Stroke: "transparent",
	ChartsCategorical09Color: "#EB7200",
	ChartsCategorical09Stroke: "transparent",
	ChartsStatusSuccessColor: "#007A4D",
	ChartsStatusSuccessStroke: "transparent",
	ChartsStatusWarningColor: "#ffbe00",
	ChartsStatusWarningStroke: "transparent",
	ChartsStatusDangerColor: "#e13212",
	ChartsStatusDangerStroke: "transparent",
	ChartsStatusNeutralColor: "#ffffff",
	ChartsStatusNeutralStroke: "#8C9CA5",
	FocusRingColor: "#0265DC",
	FocusRingColorDanger: "#e13212",
	ForegroundColor: "#141414",
	ForegroundColorSubdued: "#737475",
	ForegroundColorPrimary: "#0265DC",
	ForegroundColorDanger: "#e13212",
	ForegroundColorInverted: "#ffffff",
	ForegroundColorOnPrimary: "#ffffff",
	ForegroundColorOnDanger: "#ffffff",
	ForegroundColorOnDangerSubdued: "#bf2a00",
	ForegroundColorOnWarning: "#141414",
	ForegroundColorActivePrimary: "#004491",
	ForegroundColorActiveDanger: "#bf2a00",
	ForegroundColorHoverPrimary: "#004491",
	ForegroundColorHoverDanger: "#bf2a00",
	OverlayColorActive: "#14141429",
	OverlayColorActivePrimary: "#0265DC4D",
	OverlayColorActiveDanger: "#e132124D",
	OverlayColorActiveOn: "#14141429",
	OverlayColorActiveOnPrimary: "#14141466",
	OverlayColorActiveOnDanger: "#14141466",
	OverlayColorHover: "#14141414",
	OverlayColorHoverPrimary: "#0265DC1A",
	OverlayColorHoverDanger: "#e132121A",
	OverlayColorHoverOn: "#14141414",
	OverlayColorHoverOnPrimary: "#14141433",
	OverlayColorHoverOnDanger: "#14141433",
	ShadowColor: "#14141414",
	StatusColorInfo: "#0265DC",
	StatusColorDanger: "#e13212",
	StatusColorSuccess: "#007A4D",
	StatusColorWarning: "#ffbe00"
}
export const dark = {
	BackgroundColor: "#141414",
	BackgroundColorStrong: "#2d2e31",
	BackgroundColorStronger: "#444445",
	BackgroundColorStrongest: "#737475",
	BackgroundColorPrimary: "#78BBFA",
	BackgroundColorPrimarySubdued: "#78BBFA33",
	BackgroundColorSuccess: "#49CC93",
	BackgroundColorSuccessSubdued: "#49CC9333",
	BackgroundColorDanger: "#ff745f",
	BackgroundColorDangerSubdued: "#ff745f33",
	BackgroundColorWarning: "#ffe278",
	BackgroundColorDisabled: "#606162",
	BackgroundColorInverted: "#fafafa",
	BackgroundColorInvertedStrong: "#ffffff",
	BorderColor: "#606162",
	BorderColorSubdued: "#444445",
	BorderColorStrong: "#dfe0e1",
	BorderColorPrimary: "#78BBFA",
	BorderColorSuccess: "#49CC93",
	BorderColorDanger: "#ff745f",
	ChartsMonochrome01Color: "#8AC3FF",
	ChartsMonochrome01Stroke: "transparent",
	ChartsMonochrome02Color: "#3476FF",
	ChartsMonochrome02Stroke: "transparent",
	ChartsMonochrome03Color: "#141414",
	ChartsMonochrome03Stroke: "#8AC3FF",
	ChartsMonochrome04Color: "#F2F9FF",
	ChartsMonochrome04Stroke: "#8AC3FF",
	ChartsCategorical10Color: "#D99078",
	ChartsCategorical10Stroke: "transparent",
	ChartsCategorical01Color: "#3892F3",
	ChartsCategorical01Stroke: "transparent",
	ChartsCategorical02Color: "#3476FF",
	ChartsCategorical02Stroke: "transparent",
	ChartsCategorical03Color: "#C6F6FF",
	ChartsCategorical03Stroke: "transparent",
	ChartsCategorical04Color: "#C6B8F4",
	ChartsCategorical04Stroke: "transparent",
	ChartsCategorical05Color: "#D949A9",
	ChartsCategorical05Stroke: "transparent",
	ChartsCategorical06Color: "#ED1369",
	ChartsCategorical06Stroke: "transparent",
	ChartsCategorical07Color: "#8CC30B",
	ChartsCategorical07Stroke: "transparent",
	ChartsCategorical08Color: "#B36800",
	ChartsCategorical08Stroke: "transparent",
	ChartsCategorical09Color: "#FFA037",
	ChartsCategorical09Stroke: "transparent",
	ChartsStatusSuccessColor: "#49CC93",
	ChartsStatusSuccessStroke: "transparent",
	ChartsStatusWarningColor: "#ffe278",
	ChartsStatusWarningStroke: "transparent",
	ChartsStatusDangerColor: "#ff745f",
	ChartsStatusDangerStroke: "transparent",
	ChartsStatusNeutralColor: "#141414",
	ChartsStatusNeutralStroke: "#B4C1C8",
	FocusRingColor: "#78BBFA",
	FocusRingColorDanger: "#ff745f",
	ForegroundColor: "#ffffff",
	ForegroundColorSubdued: "#bcbcbd",
	ForegroundColorPrimary: "#78BBFA",
	ForegroundColorDanger: "#ff745f",
	ForegroundColorInverted: "#141414",
	ForegroundColorOnPrimary: "#141414",
	ForegroundColorOnDanger: "#141414",
	ForegroundColorOnDangerSubdued: "#ffffff",
	ForegroundColorOnWarning: "#141414",
	ForegroundColorActivePrimary: "#B5DEFF",
	ForegroundColorActiveDanger: "#ff745f",
	ForegroundColorHoverPrimary: "#B5DEFF",
	ForegroundColorHoverDanger: "#ff745f",
	OverlayColorActive: "#ffffff29",
	OverlayColorActivePrimary: "#78BBFA40",
	OverlayColorActiveDanger: "#f94d3240",
	OverlayColorActiveOn: "#14141429",
	OverlayColorActiveOnPrimary: "#14141429",
	OverlayColorActiveOnDanger: "#14141429",
	OverlayColorHover: "#ffffff14",
	OverlayColorHoverPrimary: "#78BBFA1A",
	OverlayColorHoverDanger: "#f94d3226",
	OverlayColorHoverOn: "#ffffff14",
	OverlayColorHoverOnPrimary: "#ffffff29",
	OverlayColorHoverOnDanger: "#ffffff29",
	ShadowColor: "#ffffff14",
	StatusColorInfo: "#78BBFA",
	StatusColorDanger: "#ff745f",
	StatusColorSuccess: "#49CC93",
	StatusColorWarning: "#ffe278"
}
