// @servicetitan/anvil2/dist/DateFieldYearless-DnVJaBBq.js
import { jsxs, jsx } from 'react/jsx-runtime';
import { Y as YearlessDateInputWithPicker } from './YearlessDateInputWithPicker-MHLlAdrH.js';
import { useRef, useState, useId } from 'react';
import { F as FieldLabel } from './FieldLabel-CHMCV9wX.js';
import { t as textFieldStyles } from './TextField.module-pD1felN8.js';
import { c as cx } from './index-tZvMCc77.js';
import { u as useLayoutPropsUtil } from './useLayoutPropsUtil-BlIWftBb.js';
import { c as childrenToString } from './childrenToString-Bz9MqbHb.js';
import { useTrackingId } from './useTrackingId.js';
import { H as Helper } from './Helper-DjWotFtO.js';

const DateFieldYearless = (props) => {
  const { layoutStyles, componentProps } = useLayoutPropsUtil(props);
  const {
    value,
    defaultValue,
    minDate,
    maxDate,
    unavailable,
    required,
    disablePicker = false,
    mode = "mm/dd",
    error,
    errorAriaLive: errorAriaLiveProp = "assertive",
    label,
    moreInfo,
    size,
    disabled,
    loading,
    labelProps,
    description,
    id: idProp,
    style,
    disableHint = false,
    onFocus,
    onBlur,
    className,
    onChange,
    ...restProps
  } = componentProps;
  const inputRef = useRef(null);
  const [openInfo, setOpenInfo] = useState(false);
  const [errorAriaLive, setErrorAriaLive] = useState("off");
  const handleFocus = (e) => {
    setOpenInfo(true);
    setErrorAriaLive(errorAriaLiveProp);
    onFocus?.(e);
  };
  const handleBlur = (e) => {
    setOpenInfo(false);
    setErrorAriaLive("off");
    onBlur?.(e);
  };
  const data = {
    label: childrenToString(props.label),
    labelProps: props.labelProps,
    description: childrenToString(props.description)
  };
  const trackingId = useTrackingId({
    name: "DateFieldYearless",
    data,
    hasOverride: !!props["data-tracking-id"]
  });
  const labelUid = useId();
  const id = idProp ?? labelUid;
  const helperUid = useId();
  const errorMessage = typeof error !== "boolean" ? error : void 0;
  const styleCombined = { ...style, ...layoutStyles };
  return /* @__PURE__ */ jsxs(
    "div",
    {
      className: cx(textFieldStyles["textfield"], className),
      "data-anv": "datefield-yearless",
      style: styleCombined,
      children: [
        label && /* @__PURE__ */ jsx(
          FieldLabel,
          {
            moreInfo,
            openMoreInfo: openInfo,
            required,
            htmlFor: id,
            ...labelProps,
            children: label
          },
          openInfo ? "open" : "closed"
        ),
        /* @__PURE__ */ jsx(
          YearlessDateInputWithPicker,
          {
            ...restProps,
            ref: inputRef,
            mode,
            required,
            minDate,
            maxDate,
            unavailable,
            disablePicker,
            size,
            error: !!error,
            disabled,
            loading,
            id,
            "data-tracking-id": trackingId,
            "aria-describedby": helperUid,
            "aria-invalid": error ? !!error : void 0,
            "aria-label": label ? void 0 : "Select a date",
            value,
            defaultValue,
            onChange,
            onFocus: handleFocus,
            onBlur: handleBlur
          }
        ),
        errorMessage || description || !disableHint && mode ? /* @__PURE__ */ jsx(
          Helper,
          {
            id: helperUid,
            hint: disableHint ? void 0 : `Format: ${mode}`,
            showCounter: false,
            errorMessage,
            errorAriaLive,
            description
          }
        ) : null
      ]
    }
  );
};
DateFieldYearless.displayName = "DateFieldYearless";

export { DateFieldYearless as D };
