// @servicetitan/anvil2/dist/indeterminate_check_box-RY9zr3xS.js
import * as React from 'react';

import './indeterminate_check_box.css';const checkbox = "_checkbox_19qga_8";
const wrapper = "_wrapper_19qga_23";
const error = "_error_19qga_103";
const label = "_label_19qga_215";
const legend = "_legend_19qga_247";
const children = "_children_19qga_259";
const checkboxStyles = {
	checkbox: checkbox,
	wrapper: wrapper,
	"required-asterisk": "_required-asterisk_19qga_37",
	"focus-label": "_focus-label_19qga_41",
	"icon-wrapper": "_icon-wrapper_19qga_47",
	"helper-wrapper": "_helper-wrapper_19qga_80",
	"icon-indeterminate": "_icon-indeterminate_19qga_85",
	"icon-unchecked": "_icon-unchecked_19qga_86",
	"icon-checked": "_icon-checked_19qga_87",
	error: error,
	"sr-only": "_sr-only_19qga_201",
	"checkbox-group": "_checkbox-group_19qga_215",
	label: label,
	"direction-row": "_direction-row_19qga_215",
	legend: legend,
	children: children
};

const SvgCheckBoxOutlineBlank = (props) => /* @__PURE__ */ React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...props }, /* @__PURE__ */ React.createElement("path", { d: "M18 19H6c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1zm1-16H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" }));

const SvgCheckBox = (props) => /* @__PURE__ */ React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...props }, /* @__PURE__ */ React.createElement("path", { d: "M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-8.29 13.29a.996.996 0 0 1-1.41 0L5.71 12.7a.996.996 0 1 1 1.41-1.41L10 14.17l6.88-6.88a.996.996 0 1 1 1.41 1.41l-7.58 7.59z" }));

const SvgIndeterminateCheckBox = (props) => /* @__PURE__ */ React.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", width: "1em", height: "1em", viewBox: "0 0 24 24", ...props }, /* @__PURE__ */ React.createElement("path", { d: "M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-3 10H8c-.55 0-1-.45-1-1s.45-1 1-1h8c.55 0 1 .45 1 1s-.45 1-1 1z" }));

export { SvgIndeterminateCheckBox as S, SvgCheckBox as a, SvgCheckBoxOutlineBlank as b, checkboxStyles as c };
