// @servicetitan/anvil2/dist/Layout-CUUb2PVr.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { c as cx } from './index-tZvMCc77.js';

import './Layout.css';const container = "_container_1suid_2";
const fluid = "_fluid_1suid_13";
const grid = "_grid_1suid_16";
const wide = "_wide_1suid_21";
const narrow = "_narrow_1suid_24";
const xs1 = "_xs1_1suid_27";
const xs2 = "_xs2_1suid_30";
const xs3 = "_xs3_1suid_33";
const xs4 = "_xs4_1suid_36";
const xs5 = "_xs5_1suid_37";
const xs6 = "_xs6_1suid_38";
const xs7 = "_xs7_1suid_39";
const xs8 = "_xs8_1suid_40";
const xs9 = "_xs9_1suid_41";
const xs10 = "_xs10_1suid_42";
const xs11 = "_xs11_1suid_43";
const xs12 = "_xs12_1suid_44";
const cell = "_cell_1suid_47";
const sm1 = "_sm1_1suid_57";
const sm2 = "_sm2_1suid_60";
const sm3 = "_sm3_1suid_63";
const sm4 = "_sm4_1suid_66";
const sm5 = "_sm5_1suid_67";
const sm6 = "_sm6_1suid_68";
const sm7 = "_sm7_1suid_69";
const sm8 = "_sm8_1suid_70";
const sm9 = "_sm9_1suid_71";
const sm10 = "_sm10_1suid_72";
const sm11 = "_sm11_1suid_73";
const sm12 = "_sm12_1suid_74";
const md1 = "_md1_1suid_149";
const md2 = "_md2_1suid_152";
const md3 = "_md3_1suid_155";
const md4 = "_md4_1suid_158";
const md5 = "_md5_1suid_161";
const md6 = "_md6_1suid_164";
const md7 = "_md7_1suid_167";
const md8 = "_md8_1suid_170";
const md9 = "_md9_1suid_173";
const md10 = "_md10_1suid_176";
const md11 = "_md11_1suid_179";
const md12 = "_md12_1suid_182";
const lg1 = "_lg1_1suid_197";
const lg2 = "_lg2_1suid_200";
const lg3 = "_lg3_1suid_203";
const lg4 = "_lg4_1suid_206";
const lg5 = "_lg5_1suid_209";
const lg6 = "_lg6_1suid_212";
const lg7 = "_lg7_1suid_215";
const lg8 = "_lg8_1suid_218";
const lg9 = "_lg9_1suid_221";
const lg10 = "_lg10_1suid_224";
const lg11 = "_lg11_1suid_227";
const lg12 = "_lg12_1suid_230";
const xl1 = "_xl1_1suid_245";
const xl2 = "_xl2_1suid_248";
const xl3 = "_xl3_1suid_251";
const xl4 = "_xl4_1suid_254";
const xl5 = "_xl5_1suid_257";
const xl6 = "_xl6_1suid_260";
const xl7 = "_xl7_1suid_263";
const xl8 = "_xl8_1suid_266";
const xl9 = "_xl9_1suid_269";
const xl10 = "_xl10_1suid_272";
const xl11 = "_xl11_1suid_275";
const xl12 = "_xl12_1suid_278";
const xxl1 = "_xxl1_1suid_287";
const xxl2 = "_xxl2_1suid_290";
const xxl3 = "_xxl3_1suid_293";
const xxl4 = "_xxl4_1suid_296";
const xxl5 = "_xxl5_1suid_299";
const xxl6 = "_xxl6_1suid_302";
const xxl7 = "_xxl7_1suid_305";
const xxl8 = "_xxl8_1suid_308";
const xxl9 = "_xxl9_1suid_311";
const xxl10 = "_xxl10_1suid_314";
const xxl11 = "_xxl11_1suid_317";
const xxl12 = "_xxl12_1suid_320";
const styles = {
	container: container,
	fluid: fluid,
	grid: grid,
	wide: wide,
	narrow: narrow,
	xs1: xs1,
	xs2: xs2,
	xs3: xs3,
	xs4: xs4,
	xs5: xs5,
	xs6: xs6,
	xs7: xs7,
	xs8: xs8,
	xs9: xs9,
	xs10: xs10,
	xs11: xs11,
	xs12: xs12,
	cell: cell,
	sm1: sm1,
	sm2: sm2,
	sm3: sm3,
	sm4: sm4,
	sm5: sm5,
	sm6: sm6,
	sm7: sm7,
	sm8: sm8,
	sm9: sm9,
	sm10: sm10,
	sm11: sm11,
	sm12: sm12,
	md1: md1,
	md2: md2,
	md3: md3,
	md4: md4,
	md5: md5,
	md6: md6,
	md7: md7,
	md8: md8,
	md9: md9,
	md10: md10,
	md11: md11,
	md12: md12,
	lg1: lg1,
	lg2: lg2,
	lg3: lg3,
	lg4: lg4,
	lg5: lg5,
	lg6: lg6,
	lg7: lg7,
	lg8: lg8,
	lg9: lg9,
	lg10: lg10,
	lg11: lg11,
	lg12: lg12,
	xl1: xl1,
	xl2: xl2,
	xl3: xl3,
	xl4: xl4,
	xl5: xl5,
	xl6: xl6,
	xl7: xl7,
	xl8: xl8,
	xl9: xl9,
	xl10: xl10,
	xl11: xl11,
	xl12: xl12,
	xxl1: xxl1,
	xxl2: xxl2,
	xxl3: xxl3,
	xxl4: xxl4,
	xxl5: xxl5,
	xxl6: xxl6,
	xxl7: xxl7,
	xxl8: xxl8,
	xxl9: xxl9,
	xxl10: xxl10,
	xxl11: xxl11,
	xxl12: xxl12
};

const LayoutElement = forwardRef(
  (props, ref) => {
    const {
      children,
      className,
      fluid = false,
      variant = "default",
      ...rest
    } = props;
    return /* @__PURE__ */ jsx("div", { className: cx(styles.container, { [styles.fluid]: fluid }), children: /* @__PURE__ */ jsx(
      "div",
      {
        className: cx(
          styles.grid,
          {
            [styles.wide]: variant === "wide",
            [styles.narrow]: variant === "narrow"
          },
          className
        ),
        ref,
        ...rest,
        children
      }
    ) });
  }
);
LayoutElement.displayName = "Layout";
const LayoutItem = forwardRef(
  ({ children, className, span = 12, xs, sm, md, lg, xl, xxl, ...props }, ref) => {
    return /* @__PURE__ */ jsx(
      "div",
      {
        ref,
        ...props,
        className: cx(
          styles.cell,
          styles[`xs${xs || span}`],
          {
            [styles[`sm${sm}`]]: sm,
            [styles[`md${md}`]]: md,
            [styles[`lg${lg}`]]: lg,
            [styles[`xl${xl}`]]: xl,
            [styles[`xxl${xxl}`]]: xxl
          },
          className
        ),
        children
      }
    );
  }
);
LayoutItem.displayName = "LayoutItem";
const Layout = Object.assign(LayoutElement, {
  Item: LayoutItem
});

export { Layout as L, LayoutElement as a, LayoutItem as b };
