// @servicetitan/anvil2/dist/LinkButton-pIUNATNE.js
import { jsx } from 'react/jsx-runtime';
import { forwardRef } from 'react';
import { l as linkStyles } from './Link.module-Cf17JWod.js';
import { useTrackingId } from './useTrackingId.js';
import { c as cx } from './index-tZvMCc77.js';
import { c as childrenToString } from './childrenToString-Bz9MqbHb.js';
import { u as useLayoutPropsUtil } from './useLayoutPropsUtil-BlIWftBb.js';

import './LinkButton.css';const styles = {
	"link-button": "_link-button_rbbxj_1"
};

const LinkButton = forwardRef(
  (props, ref) => {
    const data = {
      appearance: props.appearance,
      quiet: props.quiet,
      children: childrenToString(props.children)
    };
    const trackingId = useTrackingId({
      name: "Link",
      data,
      hasOverride: !!props["data-tracking-id"]
    });
    const { layoutStyles, componentProps } = useLayoutPropsUtil(props);
    const {
      children,
      className,
      appearance = "primary",
      quiet,
      style,
      ...rest
    } = componentProps;
    const linkClassNames = cx(
      [linkStyles["link"]],
      styles["link-button"],
      className,
      {
        [linkStyles["quiet"]]: quiet,
        [linkStyles["secondary"]]: appearance === "secondary"
      }
    );
    const styleCombined = {
      ...style,
      ...layoutStyles
    };
    return /* @__PURE__ */ jsx(
      "button",
      {
        className: linkClassNames,
        "data-anv": "link-button",
        ref,
        style: styleCombined,
        "data-tracking-id": trackingId,
        ...rest,
        children
      }
    );
  }
);
LinkButton.displayName = "LinkButton";

export { LinkButton as L };
