// @servicetitan/anvil2/dist/SelectCard-lBcnP5j5.js
import { jsx, jsxs, Fragment } from 'react/jsx-runtime';
import { useContext, createContext, useState, useId, forwardRef, useLayoutEffect } from 'react';
import { c as cx } from './index-tZvMCc77.js';
import { C as CheckboxGroup, a as Checkbox } from './Checkbox-Bhjj-jM4.js';
import { R as RadioGroup, a as Radio } from './RadioGroup-DRS2EjVH.js';
import { C as Card } from './Card-Ch3sMUo5.js';
import { u as useOptionallyControlledState } from './useOptionallyControlledState-DbDuos5L.js';

import './SelectCard.css';const SelectCardContext = createContext(null);
const useSelectCardContext = () => useContext(SelectCardContext);

const SelectCardProvider = ({
  children,
  onChange,
  selectionMode
}) => {
  const [selection, setSelection] = useState(/* @__PURE__ */ new Set());
  const radioName = useId();
  const handleChange = (e, state, options) => {
    if (!state || !state.id) {
      if (!options?.internal) {
        onChange(e, state);
      }
      return;
    }
    if (selectionMode === "single") {
      if (state.checked) {
        setSelection(/* @__PURE__ */ new Set([state.id]));
      } else {
        setSelection(/* @__PURE__ */ new Set());
      }
    } else {
      if (state.checked) {
        setSelection(selection.add(state.id));
      } else {
        selection.delete(state.id);
        setSelection(selection);
      }
    }
    if (!options?.internal) {
      onChange(e, state);
    }
  };
  const context = {
    selectionMode,
    onChange: handleChange,
    radioName,
    selection
  };
  return /* @__PURE__ */ jsx(SelectCardContext.Provider, { value: context, children });
};

const SelectCardGroup = forwardRef((props, _ref) => {
  const {
    ariaLabelledBy,
    children,
    legend,
    onChange,
    required = false,
    selectionMode,
    ...rest
  } = props;
  return /* @__PURE__ */ jsxs(SelectCardProvider, { onChange, selectionMode, children: [
    selectionMode === "multiple" && /* @__PURE__ */ jsx(
      CheckboxGroup,
      {
        "aria-labelledby": ariaLabelledBy,
        legend,
        required,
        ...rest,
        children
      }
    ),
    selectionMode === "single" && /* @__PURE__ */ jsx(
      RadioGroup,
      {
        "aria-labelledby": ariaLabelledBy,
        legend,
        required,
        ...rest,
        children
      }
    )
  ] });
});
SelectCardGroup.displayName = "SelectCardGroup";

const styles = {
	"select-card": "_select-card_3pbcy_2",
	"select-card--errored": "_select-card--errored_3pbcy_22",
	"select-card--selected": "_select-card--selected_3pbcy_25",
	"select-card--disabled": "_select-card--disabled_3pbcy_41",
	"select-card--remove-drop-shadow": "_select-card--remove-drop-shadow_3pbcy_45"
};

const SelectCardElement = forwardRef(
  (props, ref) => {
    const {
      checkboxProps,
      children,
      checked: checkedProp,
      className,
      defaultChecked: defaultCheckedProp,
      disabled,
      errored,
      radioProps,
      onChange,
      removeDropShadow,
      id,
      ...rest
    } = props;
    const [checked, setChecked] = useOptionallyControlledState({
      controlledValue: checkedProp,
      defaultValue: defaultCheckedProp || false,
      onChange
    });
    const context = useSelectCardContext();
    const [_, setForceRenderCount] = useState(0);
    useLayoutEffect(() => {
      if (defaultCheckedProp) {
        context?.onChange(
          void 0,
          {
            id,
            checked: defaultCheckedProp
          },
          { internal: true }
        );
        setForceRenderCount((prev) => prev + 1);
      }
    }, []);
    const handleChange = (event) => {
      const newChecked = event ? event.target.checked : false;
      setChecked(newChecked);
      context?.onChange(event, {
        id,
        checked: newChecked
      });
    };
    const isChecked = context ? context.selection.has(id) : checked;
    return /* @__PURE__ */ jsxs(Fragment, { children: [
      (!context || context?.selectionMode === "multiple") && /* @__PURE__ */ jsx(
        Checkbox,
        {
          ...checkboxProps,
          defaultChecked: defaultCheckedProp,
          onChange: handleChange,
          disabled,
          ref,
          label: /* @__PURE__ */ jsx(
            Card,
            {
              ...rest,
              className: cx(styles["select-card"], className, {
                [styles["select-card--remove-drop-shadow"]]: removeDropShadow,
                [styles["select-card--selected"]]: isChecked && !errored,
                [styles["select-card--errored"]]: errored,
                [styles["select-card--disabled"]]: disabled
              }),
              children
            }
          ),
          hideCheckbox: true
        }
      ),
      context?.selectionMode === "single" && /* @__PURE__ */ jsx(
        Radio,
        {
          ...radioProps,
          defaultChecked: defaultCheckedProp,
          onChange: handleChange,
          disabled,
          ref,
          name: context.radioName,
          label: /* @__PURE__ */ jsx(
            Card,
            {
              ...rest,
              className: cx(styles["select-card"], className, {
                [styles["select-card--remove-drop-shadow"]]: removeDropShadow,
                [styles["select-card--selected"]]: isChecked && !errored,
                [styles["select-card--errored"]]: errored,
                [styles["select-card--disabled"]]: disabled
              }),
              children
            }
          ),
          hideRadio: true
        }
      )
    ] });
  }
);
SelectCardElement.displayName = "SelectCard";
const SelectCard = Object.assign(SelectCardElement, {
  /**
   * SelectCardGroup component for managing groups of selectable cards.
   *
   * Features:
   * - Manages selection state for multiple SelectCard components
   * - Supports both single and multiple selection modes
   * - Provides proper accessibility with legends and ARIA labels
   * - Handles required field validation
   * - Automatic radio button grouping for single selection
   * - Context provider for child SelectCard components
   *
   * @example
   * <SelectCard.Group
   *   legend="Choose your plan"
   *   selectionMode="single"
   *   onChange={(e, state) => console.log('Selection changed:', state)}
   * >
   *   <SelectCard id="basic" onChange={() => {}}>
   *     <Text variant="headline" size="small">Basic Plan</Text>
   *   </SelectCard>
   *   <SelectCard id="premium" onChange={() => {}}>
   *     <Text variant="headline" size="small">Premium Plan</Text>
   *   </SelectCard>
   * </SelectCard.Group>
   */
  Group: SelectCardGroup
});

export { SelectCard as S, SelectCardElement as a, SelectCardGroup as b };
