// @servicetitan/anvil2/dist/useInitialFocus-Cl_Txu7W.js
import { useEffect, useState, useRef } from 'react';
import { u as useOpenCloseTransitionStates } from './useOnClickOutside-BDzuUjmY.js';
import { t as tabbable } from './DrawerContext-D4tWTLVf.js';

const ATTRIBUTE_NAME_LIST = "data-anv-dialogs";
const ATTRIBUTE_NAME_BOOL = "data-anv-dialogs-open";
const useDialogScrollLock = ({
  open,
  id,
  enableScrollChaining,
  elementRef
}) => {
  useEffect(() => {
    if (enableScrollChaining) return () => {
    };
    const root = elementRef.current?.getRootNode();
    if (!(root instanceof ShadowRoot)) return () => {
    };
    const updateDialogAttributes = (shouldAdd) => {
      const currentDialogs = root.host.getAttribute(ATTRIBUTE_NAME_LIST)?.split(" ").filter(Boolean) ?? [];
      const newDialogs = new Set(currentDialogs);
      if (shouldAdd) {
        newDialogs.add(id);
      } else {
        newDialogs.delete(id);
      }
      const dialogList = Array.from(newDialogs);
      root.host.setAttribute(ATTRIBUTE_NAME_LIST, dialogList.join(" "));
      if (dialogList.length > 0) {
        root.host.setAttribute(ATTRIBUTE_NAME_BOOL, "true");
      } else {
        root.host.removeAttribute(ATTRIBUTE_NAME_BOOL);
      }
    };
    updateDialogAttributes(open);
    return () => {
      updateDialogAttributes(false);
    };
  }, [open, id, enableScrollChaining, elementRef]);
};

const useDialogTransitionStates = (ref, openProp, options = {}) => {
  const { onOpenAnimationStart, onCloseAnimationComplete } = options;
  return useOpenCloseTransitionStates(openProp, {
    ...options,
    onOpenAnimationStart: () => {
      if (!ref.current) return;
      ref.current.showModal();
      onOpenAnimationStart?.();
    },
    onCloseAnimationComplete: () => {
      if (!ref.current) return;
      ref.current.close();
      onCloseAnimationComplete?.();
    },
    durationMs: () => {
      if (!ref.current || !window) return 0;
      const duration = window.getComputedStyle(ref.current).animationDuration;
      return parseFloat(duration) * 1e3;
    }
  });
};

const useKeyboardFocusables = (element, options = {
  observeChange: true,
  attributes: true,
  characterData: true,
  childList: true,
  subtree: true
}) => {
  const [focusables, setFocusables] = useState(null);
  const observerRef = useRef();
  useEffect(() => {
    const target = element instanceof HTMLElement ? element : element?.current;
    if (!target) return;
    if (!options.observeChange && !focusables) {
      setFocusables(tabbable(target));
      return;
    }
    const mutationCallback = () => {
      const targetArr = tabbable(target);
      if (targetArr.length === 0) return;
      setFocusables(targetArr);
    };
    const observer = new MutationObserver(mutationCallback);
    observerRef.current = observer;
    observer.observe(target, {
      childList: true,
      // Observe changes to child elements
      subtree: true,
      // Observe changes to descendants of the target
      attributes: true,
      // Observe changes to attributes
      characterData: true
      // Observe changes to text content
    });
    return () => {
      observer.disconnect();
    };
  }, [element, options, focusables]);
  return { focusables };
};

const useInitialFocus = (dialogRef, { computeFocus }) => {
  useEffect(() => {
    if (!dialogRef.current) {
      return () => {
      };
    }
    const observer = new MutationObserver((mutations) => {
      mutations.forEach((mutation) => {
        if (mutation.attributeName === "open" && mutation.target === dialogRef.current && mutation.target instanceof HTMLDialogElement && mutation.target.open === true) {
          const focusables = tabbable(dialogRef.current);
          const target = computeFocus?.(focusables) || focusables[0];
          target?.focus();
        }
      });
    });
    observer.observe(dialogRef.current, {
      attributes: true,
      attributeFilter: ["open"]
    });
    return () => {
      observer.disconnect();
    };
  }, [dialogRef, computeFocus]);
};

export { useKeyboardFocusables as a, useDialogScrollLock as b, useInitialFocus as c, useDialogTransitionStates as u };
